Add citation docs (#150)
microsoft/teams-agent-accelerator-libs-py: packages/teams_memory/README.md

You can search for memories using a topic, a natural language query, or both.

### Memory Attributions

> The memory module can retrieve attributions (citations) for any memory. Attributions are the original messages from which a memory was extracted. A single memory can have multiple attributions, as the same information may have been mentioned multiple times or combines information from multiple messages. Attributions are important because they allow users to verify the source and accuracy of memories by seeing the original messages where the information appeared.

Reviewer: ```suggestion
The memory module stores attributions (citations) for its memories. Attributions are the original messages from which a memory was extracted. A single memory can have multiple attributions, as the same information may have been mentioned multiple times or combines information from multiple messages. Attributions are important because they allow users to verify the source and accuracy of memories by seeing the original messages where the information appeared.
```
